src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1554,6 +1560,32 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                 }
             }
 
+            JObject languageWorkerSection = (JObject)config["languageWorker"];
+            value = null;
+            if (languageWorkerSection != null)
+            {
+                if (languageWorkerSection.TryGetValue("maxMessageLength", out value))
+                {
+                    var requestedGrpcMaxMessageLength = int.Parse((string)value);
+                    if (ScriptSettingsManager.Instance.IsDynamicSku)
+                    {
+                        string message = $"Cannot set {nameof(scriptConfig.MaxMessageLength)} on Consumption plan. Default MaxMessageLength: {DefaultMaxMessageLengthDynamicSku} will be used";
+                        logger?.LogWarning(message);
+                    }
+                    else
+                    {
+                        scriptConfig.MaxMessageLength = requestedGrpcMaxMessageLength * 1024 * 1024;
+                        if (scriptConfig.MaxMessageLength < 0)
+                        {
+                            // Current grpc max message limits
+                            string message = $"MaxMessageLength must be between 4MB and 2000MB.Default MaxMessageLength: {DefaultMaxMessageLength} will be used";
```

Comment:
this message describes a lower limit and upper limit but the comparison doesn't seem to match...